### PR TITLE
Added `error` operator

### PR DESCRIPTION
**Pull request by kirjorjos**

### CODE REVIEW

File: src/main/java/org/cyclops/integrateddynamics/core/evaluate/operator/Operators.java [hunk 1]
```diff
@@ -780,6 +780,17 @@ public IValue evaluate(OperatorBase.SafeVariablesGetter variables) throws Evalua
...
             ).build());
 
+    /**
+     * Throw a custom error
+     */
+    public static final IOperator STRING_ERROR  = REGISTRY.register(OperatorBuilders.STRING_2.symbol("error").operatorName("string_error")
+            .inputType(ValueTypes.STRING).renderPattern(IConfigRenderPattern.SUFFIX_1_LONG)
+            .function(
+                (variables) -> {
+                    throw new EvaluationException(Component.translatable(variables.getValue(0, ValueTypes.STRING).getRawValue()));
```
Comment (rubensworks) [inline src/main/java/org/cyclops/integrateddynamics/core/evaluate/operator/Operators.java:790]: Is the `translatable` intentional here?
Could make sense to have that here indeed.

File: src/main/resources/assets/integrateddynamics/lang/en_us.json [hunk 2]
```diff
@@ -1060,6 +1060,8 @@
     "operator.integrateddynamics.string.name.info": "Get the string or name of a named object or converts it to a string",
     "operator.integrateddynamics.string.unique_name": "Unique Name",
     "operator.integrateddynamics.string.unique_name.info": "Get the unique name of an object",
+	"operator.integrateddynamics.string.string_error": "Error",
```
Comment (rubensworks) [inline src/main/resources/assets/integrateddynamics/lang/en_us.json]: Indentation seems to have gone wrong here. Could you use spaces instead of tabs?

**Review by rubensworks — CHANGES_REQUESTED** [review]
Thanks for this @kirjorjos!

Could you also add some unit tests for this new operator in https://github.com/CyclopsMC/IntegratedDynamics/blob/d7b38c774b591aa357ff76d3d539a718fe6167ab/src/test/java/org/cyclops/integrateddynamics/core/evaluate/variable/TestStringOperators.java ?

The CLA check probably fails because your git user that created the commit is not linked to your github account. (you can see this, as there is no link behind your commit's username)

File: src/main/java/org/cyclops/integrateddynamics/core/evaluate/operator/Operators.java (hunk 1 above)
Comment (kirjorjos) [reply src/main/java/org/cyclops/integrateddynamics/core/evaluate/operator/Operators.java:790]: I saw translatable getting used in other parts of the file to make a `MutableComponent` that the `EvaluationException` constructor expects, but if there's another way you'd prefer me make one, I can definitely change it.

**Comment by kirjorjos** [comment]
I think I added the lang file indentation fix and tests for the operator, I'm not familiar with github enough to be 100% sure though if I'm being honest.

**Comment by rubensworks** [comment]
You can fix your git settings like this: https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address

<blockquote><div> GitHub Docs</div><div><strong><a href="https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address">Setting your commit email address - GitHub Docs</a></strong></div><div>You can set the email address that is used to author commits on GitHub and on your computer.</div></blockquote>

**Comment by kirjorjos** [comment]
Thank you for the documentation on fixing it, had to do some googling to retroactively change the email the commit used, but it's all working now and future ones will have the correct one.

File: src/test/java/org/cyclops/integrateddynamics/core/evaluate/variable/TestStringOperators.java [hunk 3]
```diff
@@ -780,4 +780,18 @@ public void testInvalidInputTypeNamedName() throws EvaluationException {
         Operators.NAMED_NAME.evaluate(new IVariable[]{DUMMY_VARIABLE});
     }
 
+    /**
+     * ----------------------------------- STRING_ERROR -----------------------------------
+     */
+
+    @Test(expected = EvaluationException.class)
+    public void testSymbolsError() throws EvaluationException {
```
Comment (rubensworks) [inline src/test/java/org/cyclops/integrateddynamics/core/evaluate/variable/TestStringOperators.java]:
```suggestion
    public void testStringErrorWithSymbols() throws EvaluationException {
```

File: src/test/java/org/cyclops/integrateddynamics/core/evaluate/variable/TestStringOperators.java [hunk 4]
```diff
@@ -780,4 +780,18 @@ public void testInvalidInputTypeNamedName() throws EvaluationException {
...
+        Operators.STRING_ERROR.evaluate(sregex);
+    }
+
+    @Test(expected = EvaluationException.class)
+    public void testSpacesError() throws EvaluationException {
```
Comment (rubensworks) [inline src/test/java/org/cyclops/integrateddynamics/core/evaluate/variable/TestStringOperators.java]:
```suggestion
    public void testStringErrorWithSpaces() throws EvaluationException {
```

File: src/test/java/org/cyclops/integrateddynamics/core/evaluate/variable/TestStringOperators.java [hunk 5]
```diff
@@ -780,4 +780,18 @@ public void testInvalidInputTypeNamedName() throws EvaluationException {
...
+        Operators.STRING_ERROR.evaluate(shelloWorld);
+    }
+
```
Comment (rubensworks) [inline src/test/java/org/cyclops/integrateddynamics/core/evaluate/variable/TestStringOperators.java:796]: Could you also add tests for checking too many, too few, and incorrectly typed inputs? (just like [the other tests](https://github.com/CyclopsMC/IntegratedDynamics/blob/0d84dfeea8d48357d929f4fdf1e6fc5cfc75070c/src/test/java/org/cyclops/integrateddynamics/core/evaluate/variable/TestStringOperators.java#L158-L171))

**Review by rubensworks — CHANGES_REQUESTED** [review]

**Comment by kirjorjos** [comment]
oh, looking at the comparison, also fixed a typo in the concat tests and promptly forgot, if you need me to, I'll change it back, but I don't see anything where it should mater

**Comment by rubensworks** [comment]
Super, thanks @kirjorjos!

**Comment by rubensworks** [comment]
@kirjorjos I just noticed you submitted this PR to `master-1.19`, while it's actually `master-1.19-lts` that is being supported atm. Could you open a new PR for that one? `master-1.19` is EOL.

**Pull request merged**